libass/ass_font.c

```diff
@@ -35,6 +36,159 @@
 #include "ass_utils.h"
 #include "ass_shaper.h"
 
+/**
+ *  Get the mbcs codepoint from the output bytes of iconv/WideCharToMultiByte,
+ *  by treating the bytes as a prefix-zero-byte-omitted big-endian integer.
+ */
+static inline uint32_t pack_mbcs_bytes(const char *bytes, size_t length)
+{
+    uint32_t ret = 0;
+    for (size_t i = 0; i < length; ++i) {
+        ret <<= 8;
+        ret |= (uint8_t) bytes[i];
+    }
+    return ret;
+}
+
+/**
+ * Convert a UCS-4 code unit to a packed uint32_t in given multibyte encoding
+ */
+static uint32_t convert_unicode_to_mb(FT_Encoding encoding, uint32_t codepoint);
+
```

Comment:
I'm not too fond of the declaration outside the `#if`-blocks added by the latest change.
At one hand it is unnecessary, because we define the static function in all cases directly below the declaration,
on the other hand, if this weren't the case, adding a unconditional declaration and conditional definitions would delay errors due to a missing definition to the linker stage *(we use `-Werror-implicit-function-declaration`)*.

If you want to preserve the doc comment for `convert_unicode_to_mb` maybe merge it in the comment preceding the `#if defined(_WIN32)`, eg similar to this:
```C
/**
 * Convert a UCS-4 code unit to a packed uint32_t in given MS-flavour multibyte encoding
 *
 * We don't exclude Cygwin from _WIN32, since we only use WideCharToMultiByte which 
 * shouldn't violate any Cygwin restrictions and unlike iconv is guaranteed to be MS-flavour
 */
```